src/Tizen.NUI.WindowSystem/src/public/InputGesture.cs

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright(c) 2023 Samsung Electronics Co., Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+using System;
+using System.ComponentModel;
+using static Tizen.NUI.WindowSystem.Interop.InputGesture;
+
+namespace Tizen.NUI.WindowSystem
+{
+    /// <summary>
+    /// Class for the Tizen Input Gesture.
+    /// </summary>
+    /// <privilege>
+    /// http://tizen.org/privilege/gesturegrab
+    /// </privilege>
+    /// This class is need to be hidden as inhouse API.
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class InputGesture : IDisposable
+    {
+        private IntPtr _handler;
+        private bool disposed = false;
+        private bool isDisposeQueued = false;
+        IntPtr edgeSwipeG;
+        IntPtr edgeDragG;
+        IntPtr tapG;
+        IntPtr palmG;
+
+        private static event EventHandler<EdgeSwipeEventArgs> _edgeSwipeEventHandler;
+        private static EdgeSwipeCb _edgeSwipeDelegate;
+
+        private static event EventHandler<EdgeDragEventArgs> _edgeDragEventHandler;
+        private static EdgeDragCb _edgeDragDelegate;
+
+
+        private static event EventHandler<TapEventArgs> _tapEventHandler;
+        private static TapCb _tapDelegate;
+
+        private static event EventHandler<PalmCoverEventArgs> _palmCoverEventHandler;
+        private static PalmCoverCb _palmCoverDelegate;
+
+        /// <summary>
+        /// Enumeration of gesture modes.
+        /// </summary>
+        public enum GestureMode
```

Comment:
❌ DO NOT use public nested types as a logical grouping construct; use namespaces for this.

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types